### PR TITLE
Add predicate token filter docs #8272

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Add predicate token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8272

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _analyzers/token-filters/index.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ Normalization | `arabic_normalization`: [ArabicNormalizer](https://lucene.apache
 `pattern_replace` | N/A | Matches a pattern in the provided regular expression and replaces matching substrings. Uses [Java regular expression syntax](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html).
 `phonetic` | N/A | Uses a phonetic encoder to emit a metaphone token for each token in the token stream. Requires installing the `analysis-phonetic` plugin.
 `porter_stem` | [PorterStemFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/en/PorterStemFilter.html) | Uses the [Porter stemming algorithm](https://tartarus.org/martin/PorterStemmer/) to perform algorithmic stemming for the English language.
-`predicate_token_filter` | N/A | Removes tokens that don’t match the specified predicate script. Supports inline Painless scripts only.
+[`predicate_token_filter`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/predicate-token-filter/) | N/A | Removes tokens that don’t match the specified predicate script. Supports inline Painless scripts only.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:50]:
```suggestion
[`predicate_token_filter`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/predicate-token-filter/) | N/A | Removes tokens that do not match the specified predicate script. Supports only inline Painless scripts.
```

File: _analyzers/token-filters/predicate-token-filter.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+---
+layout: default
+title: Predicate token filter
+parent: Token filters
+nav_order: 340
+---
+
+# Predicate token filter
+
+The `predicate_token_filter` evaluates whether tokens should be kept or discarded, depending on the conditions defined in a custom script. The tokens are evaulated in the analysis predicate context. This filter supports only inline Painless scripts.
```
Comment (natebower) [inline _analyzers/token-filters/predicate-token-filter.md:10]:
```suggestion
The `predicate_token_filter` evaluates whether tokens should be kept or discarded, depending on the conditions defined in a custom script. The tokens are evaluated in the analysis predicate context. This filter supports only inline Painless scripts.
```

File: _analyzers/token-filters/predicate-token-filter.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+---
+layout: default
+title: Predicate token filter
+parent: Token filters
+nav_order: 340
+---
+
+# Predicate token filter
+
```
Comment (natebower) [inline _analyzers/token-filters/predicate-token-filter.md:9]: Line 10, second sentence: "in the analysis predicate context" => "in the context of the analysis predicate"?

File: _analyzers/token-filters/predicate-token-filter.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+The `predicate_token_filter` evaluates whether tokens should be kept or discarded, depending on the conditions defined in a custom script. The tokens are evaulated in the analysis predicate context. This filter supports only inline Painless scripts.
+
+## Parameters
+
+The `predicate_token_filter` has one required parameter: `script`. This parameter provides a condition that is used to evaluate if the token should be kept.
```
Comment (natebower) [inline _analyzers/token-filters/predicate-token-filter.md:14]:
```suggestion
The `predicate_token_filter` has one required parameter: `script`. This parameter provides a condition that is used to evaluate whether the token should be kept. 
```

File: _analyzers/token-filters/predicate-token-filter.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+The `predicate_token_filter` evaluates whether tokens should be kept or discarded, depending on the conditions defined in a custom script. The tokens are evaulated in the analysis predicate context. This filter supports only inline Painless scripts.
+
+## Parameters
+
+The `predicate_token_filter` has one required parameter: `script`. This parameter provides a condition that is used to evaluate if the token should be kept. 
+
+## Example
+
+The following example request creates a new index named `predicate_index` and configures an analyzer with a `predicate_token_filter`. The filter specifies to output tokens only if they are longer than 7 characters:
```
Comment (natebower) [inline _analyzers/token-filters/predicate-token-filter.md:18]:
```suggestion
The following example request creates a new index named `predicate_index` and configures an analyzer with a `predicate_token_filter`. The filter specifies to only output tokens if they are longer than 7 characters:
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws One comment and several changes. Thanks!

File: _analyzers/token-filters/predicate-token-filter.md (hunk 3 above)
Comment (kolchfa-aws) [reply _analyzers/token-filters/predicate-token-filter.md:9]: Analysis predicate context is the proper term

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**